src/Symfony/Bundle/TwigBundle/Controller/ExceptionController.php

```diff
@@ -50,25 +60,61 @@ public function __construct(Environment $twig, bool $debug)
      *
      * @return Response
      *
-     * @throws \InvalidArgumentException When the exception template does not exist
+     * @deprecated since Symfony 4.4
      */
-    public function showAction(Request $request, FlattenException $exception, DebugLoggerInterface $logger = null)
+    public function showAction(Request $request, FlattenException $exception)
     {
-        $currentContent = $this->getAndCleanOutputBuffering($request->headers->get('X-Php-Ob-Level', -1));
-        $showException = $request->attributes->get('showException', $this->debug); // As opposed to an additional parameter, this maintains BC
-
-        $code = $exception->getStatusCode();
-
-        return new Response($this->twig->render(
-            (string) $this->findTemplate($request, $request->getRequestFormat(), $code, $showException),
-            [
-                'status_code' => $code,
-                'status_text' => isset(Response::$statusTexts[$code]) ? Response::$statusTexts[$code] : '',
-                'exception' => $exception,
-                'logger' => $logger,
-                'currentContent' => $currentContent,
-            ]
-        ), 200, ['Content-Type' => $request->getMimeType($request->getRequestFormat()) ?: 'text/html']);
+        @trigger_error(sprintf('The "%s()" method is deprecated since Symfony 4.4, use "__invoke" method instead.', __METHOD__), E_USER_DEPRECATED);
+
+        return new Response($this->render($exception, $request), 200, [
+            'Content-Type' => $request->getMimeType($request->getRequestFormat()),
+        ]);
+    }
+
+    /**
+     * Converts an Exception to a Response.
+     *
+     * A "showException" request parameter can be used to force display of an error page (when set to false) or
+     * the exception page (when true). If it is not present, the "debug" value passed into the constructor will
+     * be used.
+     *
+     * @return Response
+     */
+    public function __invoke(Request $request, FlattenException $exception)
+    {
+        $request->attributes->set('_is_legacy', false);
+
+        return new Response($this->render($exception, $request), 200, [
+            'Content-Type' => $request->getMimeType($request->getRequestFormat()),
+        ]);
+    }
+
+    protected function render(FlattenException $exception, Request $request): string
+    {
+        $statusCode = $exception->getStatusCode();
+        $showException = $request->attributes->get('showException', $this->debug);
+
+        $template = $this->findTemplate($request, $request->getRequestFormat(), $statusCode, $showException);
+        if (null === $template) {
+            try {
+                // fallback to the native renderer system
+                return $this->errorFormatter->render($exception, $request->getRequestFormat());
+            } catch (ErrorRendererNotFoundException $e) {
+                $request->setRequestFormat('html');
```

Comment:
The request format should not be changed. I know it was there before but to me that is broken. Just because I don't have an error response in a specifc format, doesn't change the __request__ format. The response content-type should be set independently.

Reply:
If I understand correctly, are you suggesting don't catch the `ErrorRendererNotFoundException` exception instead? thus, forcing you to create one... that would make sense to me.

Reply:
No, falling back to HTML which must exist (because it's provided by default) is good because it's faul-tolerant. What I want is to not call `$request->setRequestFormat('html');` in this case.